examples/trigger-phrase/agent.py

```diff
@@ -0,0 +1,166 @@
+
+    async def _synth_task():
+        async for ev in tts_stream:
+            playout_q.put_nowait(ev.frame)
+
+        playout_q.put_nowait(None)
+
+    synth_task = asyncio.create_task(_synth_task())
+    playout_task = asyncio.create_task(_playout_task(playout_q, agent_audio_source))
+
+    async for ev in stt_stream:
+        if ev.type == stt.SpeechEventType.FINAL_TRANSCRIPT:
+
+            new_transcribed_text_words = word_tokenizer_without_punctuation.tokenize(
+                text=ev.alternatives[0].text
+            )
+            for i in range(len(trigger_phrase_words)):
+                if (
+                    len(new_transcribed_text_words) < len(trigger_phrase_words)
+                    or trigger_phrase_words[i].lower()
+                    != new_transcribed_text_words[i].lower()
+                ):
+                    # ignore user speech by not sending it to LLM
+                    break
+                elif i == len(trigger_phrase_words) - 1:
+                    # trigger phrase is validated
+                    new_chat_context = llm_stream.chat_ctx.append(
+                        text=ev.alternatives[0].text
+                    )
+                    llm_stream = local_llm.chat(chat_ctx=new_chat_context)
+                    llm_reply_stream = _llm_stream_to_str_iterable(llm_stream)
+                    async for seg in llm_reply_stream:
+                        tts_stream.push_text(seg)
+                    tts_stream.flush()
+    await asyncio.gather(synth_task, playout_task)
+    await tts_stream.aclose()
+
+
+async def _llm_stream_to_str_iterable(stream: llm.LLMStream) -> AsyncIterable[str]:
+    async for chunk in stream:
+        content = chunk.choices[0].delta.content
+        if content is None:
+            continue
+        yield content
+
+
+async def entrypoint(ctx: JobContext):
+    logger.info("starting trigger-phrase agent example")
+
+    vad = silero.VAD.load(
+        min_speech_duration=0.01,
+        min_silence_duration=0.5,
+    )
+
+    stt_local = stt.StreamAdapter(
+        stt=deepgram.STT(keywords=[(trigger_phrase, 3.5)]), vad=vad
+    )
+    stt_stream = stt_local.stream()
+
+    await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)
+    first_participant = await ctx.wait_for_participant()
+
+    # publish agent track
+    tts = elevenlabs.TTS(model_id="eleven_turbo_v2")
+    agent_audio_source = rtc.AudioSource(tts.sample_rate, tts.num_channels)
+    agent_track = rtc.LocalAudioTrack.create_audio_track(
+        "agent-mic", agent_audio_source
+    )
+    options = rtc.TrackPublishOptions()
+    options.source = rtc.TrackSource.SOURCE_MICROPHONE
+    publication = await ctx.room.local_participant.publish_track(agent_track, options)
+
+    # setup LLM
+    initial_ctx = llm.ChatContext().append(
+        role="system",
+        text=(
+            f"You are {trigger_phrase}, a voice assistant created by LiveKit. Your interface with users will be voice. "
```

Comment:
weird misleading use of trigger_phrase here. this implies that it can only be used as a name, i think it's best to drop it

Reply:
yeah I agree. I was also debating about this but my thought was that it might be helpful to give the LLM a bit more context